src/event-timer.ts

```diff
@@ -161,20 +179,21 @@ export class EventTimer {
 				},
 			],
 		};
+		const localWindow = window as unknown as LocalWindow;
```

Comment:
Bumping TypeScript from v4.7.4 to v4.8.4 means we lose the `NavigatorNetworkInformation` type as it is not supported by the majority of browsers.

Not so happy about this fix. Overriding the window.navigator type does not seem possible as you then run into:

`Subsequent property declarations must have the same type` errors for `lib.dom.d.ts`

Is there a better solution?

Reply:
Could you try overwriting the `Navigator` type in the global scope directly? Something like

```typescript
declare global {
	interface Navigator {
		readonly connection: NetworkInformation;
	}
}
```

Reply:
Nice idea thanks! I've tried it out but that still doesn't override `Navigator` from `lib.dom.d.ts` and you get a similar error message:

`All declarations of 'navigator' must have identical modifiers`

I shall pick your TS brain IRL :)

Reply:
I've got a [branch here](https://github.com/guardian/commercial-core/compare/test-navigator-types) with a minimal reproduction, but am not too sure it's exactly what we're after, so happy to discuss!